packages/key-management/src/restoreKeyAgent.ts

```diff
@@ -51,6 +52,10 @@ export async function restoreKeyAgent<T extends SerializableKeyAgentData>(
   dependencies: KeyAgentDependencies,
   getPassword?: GetPassword
 ): Promise<KeyAgent> {
+  if (data.knownAddresses && Symbol.iterator in data.knownAddresses) {
```

Comment:
It's a non-nullable array, why is this check needed?
```suggestion
```

Reply:
That's a problem with the test, not the implementation. Just need to add `knownAddresses: []` there

Reply:
Sure it's possible, but we're trusting that users will comply with our types everywhere else. This is only happening because of `as any`.